src/library/scanner/recursivescandirectorytask.cpp

```diff
@@ -50,9 +53,9 @@ void RecursiveScanDirectoryTask::run() {
     QRegularExpression supportedCoverExtensionsRegex =
             m_scannerGlobal->supportedCoverExtensionsRegex();
 
-    while (it.hasNext()) {
-        QString currentFile = it.next();
-        QFileInfo currentFileInfo = it.fileInfo();
+    for (auto it = children.begin(); it != children.end(); ++it) {
```

Comment:
This can become a range based loop
```suggestion
    for (const auto& child : children) {
```